modopt/opt/algorithms/base.py

```diff
@@ -34,6 +34,28 @@ class SetUp(Observable):
     use_gpu : bool, optional
         Option to use available GPU
 
+    Notes
+    -----
+    If provided, the ``metrics`` argument should be a nested dictionary of the
+    following form::
+
+        metrics = {
+            "metric_name": {
+                "metric": callable,
+                "mapping": {"x_new": "test"},
+                "cst_kwargs": {"ref": ref_image},
+                "early_stopping": False,
+            }
+        }
+
+    Where `callable` is a function with arguments being for instance
+    `test` and `ref`. The mapping of the argument uses the same keys as the
+    output of  `get_notify_observer_kwargs`, `cst_kwargs` defines constant
+    arguments that will always be passed to the metric call.
+    If `early_stopping` is True, the metric will be used to check for
+    convergence of the algorithm, in that case it is recommended to have
+    ``metric_call_period = 1``
```

Comment:
Great, but the variable names need double backticks to be rendered as literals in the HTML.